web/cobrands/fixmystreet/fixmystreet.js

```diff
@@ -1049,6 +1049,239 @@ $.extend(fixmystreet.set_up, {
+            fixmystreet.maps.markers_highlight( id );
+        }).on('mouseleave', function(){
+            timeout = setTimeout( fixmystreet.maps.markers_highlight, 50 );
+        });
+
+        fixmystreet.set_up.expandable_list_items();
+
+        // Add a "select this report" button, when on the report inspect form.
+        if ( $('#report_inspect_form').length ) {
+            $reports.each(function(){
+                var $button = $('<button>').addClass('btn btn--small btn--primary');
+                $button.text(translation_strings.this_report);
+                $button.on('click', function(e) {
+                    e.preventDefault(); // Prevent button from submitting parent form
+                    var report_id = $(this).closest('li').data('reportId');
+                    $('#report_inspect_form [name="duplicate_of"]').val(report_id);
+                    $(this).closest('li')
+                        .addClass('item-list__item--selected')
+                        .siblings('.item-list__item--selected')
+                        .removeClass('item-list__item--selected');
+                });
+                $(this).find('.item-list__item--expandable__actions').append($button);
+            });
+        }
+
+        // Add a "track this report" button when on the regular reporting form.
+        if ( $('#problem_form').length ) {
+            $reports.each(function() {
+                var id = parseInt( $(this).data('reportId'), 10 );
+                var alert_url = '/alert/subscribe?id=' + encodeURIComponent(id);
+                var $button = $('<a>').addClass('btn btn--small btn--primary btn--get-updates');
+                $button.text(translation_strings.get_updates);
+                $button.attr('href', alert_url);
+                $(this).find('.item-list__item--expandable__actions').append($button);
+            });
+        }
+    }
+
+    function render_duplicate_pins(api_response) {
+        var markers = fixmystreet.maps.markers_list( api_response.pins, true );
+        fixmystreet.markers.removeFeatures( current_duplicate_markers );
+        fixmystreet.markers.addFeatures( markers );
+        current_duplicate_markers = markers;
+    }
+
+    function remove_duplicate_list(cb) {
+        var animations = [];
+
+        animations.push( $.Deferred() );
+        $('#js-duplicate-reports').slideUp(function(){
+            $(this).addClass('hidden');
+            $(this).find('ul').empty();
+            animations[0].resolve();
+        });
+        if ( $('#problem_form').length ) {
+            animations.push( $.Deferred() );
+            $('.js-hide-if-invalid-category').slideDown(function(){
+                animations[1].resolve();
+            });
+        }
+
+        $.when.apply(this, animations).then(cb);
+    }
+
+    function remove_duplicate_pins() {
+        fixmystreet.markers.removeFeatures( current_duplicate_markers );
+    }
+
+    function inspect_form_state_change() {
```

Comment:
This function (and any related staff-only bits) could remain in staff.js, if it's only applicable to staff...